pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -0,0 +1,1097 @@
+        _prankStaticCall();
+        (
+            uint256[] memory queryPathAmountsOut,
+            address[] memory queryTokensOut,
+            uint256[] memory queryAmountsOut
+        ) = batchRouter.querySwapExactIn(paths, bytes(""));
+        vm.revertTo(snapshotId);
+
+        // Measure tokens before actual swap
+        SwapResultLocals memory vars = _createSwapResultLocals(
+            SwapKind.EXACT_IN,
+            IERC4626(address(paths[0].steps[0].tokenOut)),
+            IERC4626(address(paths[0].steps[1].tokenOut))
+        );
+        vars.expectedDeltaTokenIn = paths[0].exactAmountIn;
+
+        if (paths[0].tokenIn == _token1Fork) {
+            uint256 wrappedAmountIn = _previewDepositWithBuffer(ybToken1, vars.expectedDeltaTokenIn, shouldUseConvert);
+            uint256 wrappedAmountInScaled18 = wrappedAmountIn.divDown(_ybToken1Factor);
+            // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
+            uint256 wrappedAmountOutRaw = wrappedAmountInScaled18.mulDown(_ybToken2Factor);
+            vars.expectedDeltaTokenOut = _previewRedeemWithBuffer(ybToken2, wrappedAmountOutRaw, shouldUseConvert);
+        } else {
+            uint256 wrappedAmountIn = _previewDepositWithBuffer(ybToken2, vars.expectedDeltaTokenIn, shouldUseConvert);
+            uint256 wrappedAmountInScaled18 = wrappedAmountIn.divDown(_ybToken2Factor);
+            // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
+            uint256 wrappedAmountOutRaw = wrappedAmountInScaled18.mulDown(_ybToken1Factor);
+            vars.expectedDeltaTokenOut = _previewRedeemWithBuffer(ybToken1, wrappedAmountOutRaw, shouldUseConvert);
+        }
+
+        vars.expectedBufferDeltaTokenIn = expectedBufferDeltaTokenIn;
+        vars.expectedBufferDeltaTokenOut = expectedBufferDeltaTokenOut;
+
+        vm.prank(lp);
+        (
+            uint256[] memory actualPathAmountsOut,
+            address[] memory actualTokensOut,
+            uint256[] memory actualAmountsOut
+        ) = batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
+
+        // The error is proportional to the amount of decimals of token in and token out. If tokenIn has 6 decimals
+        // and tokenOut has 18 decimals, an error of 1 wei in amountOut of the first buffer generates an error in the
+        // order of 1e12 (1e18/1e6) in amountOut of the last buffer.
+        // But, if it's the opposite case, 1e6/1e18 is rounded to 0, but the max error is actually 1 (the error in the
+        // tokenOut token itself), so the division is incremented by 1.
+        // Finally, the error can be slightly bigger than the division of the factors, since each token has a rate. The
+        // error is multiplied by 2 to give some space for rounding errors related to the rates.
+        uint256 tolerance = (2 *
+            (paths[0].tokenIn == _token1Fork ? _token2Factor / _token1Factor : _token1Factor / _token2Factor)) + 1;
```

Comment:
Alright, fair enough.

Can we also assert that the relative error is < 0.1%, or 0.01%? I'd keep this absolute tolerance as well, but in practice this will also depend on the pool type we're using I suppose.